tools/image_processing/imagej2/imagej2_adjust_threshold_binary_jython_script.py

```diff
@@ -23,28 +21,23 @@
 input_image_plus_copy = input_image_plus.duplicate()
 image_processor_copy = input_image_plus_copy.getProcessor()
 
-# Convert image to binary if necessary.
-if not image_processor_copy.isBinary():
-    # Convert the image to binary grayscale.
-    IJ.run(
-        input_image_plus_copy,
-        "Make Binary",
-        "iterations=1 count=1 edm=Overwrite do=Nothing",
-    )
-# Set the options.
-if black_background:
-    method_str = "%s dark" % method
-else:
-    method_str = method
-IJ.setAutoThreshold(input_image_plus_copy, method_str)
+if method != "Manual":
+    # Set the options.
+    if black_background:
+        method_str = "%s dark" % method
+    else:
+        method_str = method
+    threshold_min = 1
+    threshold_max = float("inf")
```

Comment:
I think this is the good values of thresholding to highlight what have been set as mask but I would prefer that someone checks it carefully.

Reply:
May I suggest this, is it possible?

```python
# Set threshold to exclude pure black (background) but capture even low-intensity features
threshold_min = 1  # Starting point to exclude pure black

# Set threshold_max based on image bit-depth
# For 8-bit images, use 255; for 16-bit images, use 65535
if bit_depth == 8:
    threshold_max = 255  # Default for 8-bit images
elif bit_depth == 16:
    threshold_max = 65535  # Default for 16-bit images
else:
    threshold_max = float('inf')  # General fallback if bit depth is unknown

# The values can be adjusted by the user based on image characteristics

Reply:
Sure, I implement it.